tests/CMakeLists.txt

```diff
@@ -25,3 +25,16 @@ add_test(mutex_is_actually_mutex_test ${TEST_BINARY_NAME} mutex_is_actually_mute
 add_test(high_res_clock_increments_test ${TEST_BINARY_NAME} high_res_clock_increments_test)
 add_test(sys_clock_increments_test ${TEST_BINARY_NAME} sys_clock_increments_test)
 add_test(error_code_cross_thread_test, ${TEST_BINARY_NAME} error_code_cross_thread_test)
+add_test(array_list_order_push_back_pop_front_test ${TEST_BINARY_NAME} array_list_order_push_back_pop_front_test)
+add_test(array_list_order_push_back_pop_back_test ${TEST_BINARY_NAME} array_list_order_push_back_pop_back_test)
+add_test(array_list_exponential_mem_model_test ${TEST_BINARY_NAME} array_list_exponential_mem_model_test)
+add_test(array_list_iteration_test ${TEST_BINARY_NAME} array_list_iteration_test)
+add_test(array_list_preallocated_iteration_test ${TEST_BINARY_NAME} array_list_preallocated_iteration_test)
+add_test(array_list_preallocated_push_test ${TEST_BINARY_NAME} array_list_preallocated_push_test)
+add_test(array_list_shrink_to_fit_test ${TEST_BINARY_NAME} array_list_shrink_to_fit_test)
+add_test(array_list_shrink_to_fit_static_test ${TEST_BINARY_NAME} array_list_shrink_to_fit_static_test)
+add_test(array_list_clear_test ${TEST_BINARY_NAME} array_list_clear_test)
+add_test(array_list_copy_test ${TEST_BINARY_NAME} array_list_copy_test)
+add_test(array_list_not_enough_space_test ${TEST_BINARY_NAME} array_list_not_enough_space_test)
+add_test(array_list_not_enough_space_test_failure ${TEST_BINARY_NAME} array_list_not_enough_space_test_failure)
+
```

Comment:
Unfortunately, github refused to load the diff for array_list_test.c, so I can't leave comments there directly, but some observations:

1. A lot of the message strings seem really redundant. Can we just add a simple `ASSERT_SUCCESS(foo())` macro, which will auto-format `test_foo.c:321: foo() failed unexpectedly due to error code 1234 (Error String)`? Same for ASSERT_*_EQUALS, most of them say the same thing as their parameters, but with more words. Having a specific string can be occasionally useful - but the majority of cases, the arguments are self-descriptive.
2. Are we leaking memory if the tests fail?

Reply:
1. Yes I've noticed that, I'd be game for having ASSERTS that don't take a format string.
2. Sort of and I think that's fine compared to having to worry about tests with cleanup blocks on failure. If the tests fail, I don't really care if the memchecks fail. Where I want them is when the tests pass logically, I want the build to fail if there's a leak.

However, the  before/after hooks allow for a more C++ fixtures approach, if that's what you want, and there the cleanup could be done regardless of pass/fail status.